packages/go_router/lib/src/delegate.dart

```diff
@@ -278,8 +279,20 @@ class ImperativeRouteMatch extends RouteMatch {
     required super.error,
     required super.pageKey,
     required this.matches,
-  });
+  }) : _completer = Completer<T?>();
 
   /// The matches that produces this route match.
   final RouteMatchList matches;
+
+  /// The completer for the future returned by [GoRouter.push].
+  final Completer<T?> _completer;
+
+  @override
+  void complete([dynamic value]) {
+    _completer.complete(value as T?);
+  }
+
+  /// The future of the [RouteMatch] completer. When the future completes, this
```

Comment:
```suggestion
  /// The future of the [RouteMatch] completer.
  ///
  /// When the future completes, this
```

Reply:
Didn't know about private methods being accessed if in the same file, really cool!